pygmt/src/grdcut.py

```diff
@@ -102,14 +104,18 @@ def grdcut(grid, **kwargs):
     >>> # and a y-range of 21 to 24
     >>> new_grid = pygmt.grdcut(grid=grid, region=[12, 15, 21, 24])
     """
-    with GMTTempFile(suffix=".nc") as tmpfile:
-        with Session() as lib:
-            file_context = lib.virtualfile_from_data(check_kind="raster", data=grid)
-            with file_context as infile:
-                if (outgrid := kwargs.get("G")) is None:
-                    kwargs["G"] = outgrid = tmpfile.name  # output to tmpfile
-                lib.call_module(
-                    module="grdcut", args=build_arg_string(kwargs, infile=infile)
-                )
+    with Session() as lib:
```

Comment:
Some explanations for the new `grdcut` codes:

- Lines 108-110: Create two virtual files, one for the input `grid`, another one for the out grid
- Lines 111-114: Set `-G{outfile}` so that the `grdcut` call will store the result grid in the virtualfile `outfile`
- Lines 115: Read the content of the virtual file to the variable `gmtgrid`
- Lines 117-121: Process the result `gmtgrid` based on the value of `outgrid`.
  - if `outgrid` is not None, then write it to a grid file by calling the special `write` module
  - if `outgrid` is None, then convert the `GMT_GRID` object to a `xarray.DataArray` object.